cdap-ui/app/cdap/components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/RunTabs/AllRuns.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import PropTypes from 'prop-types';
+import React, {PureComponent} from 'react';
+import RuntimeArgsTabContent from 'components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/RuntimeArgsTabContent';
+import {updatePreferences, runPipeline} from 'components/PipelineConfigurations/Store/ActionCreator';
+import BtnWithLoading from 'components/BtnWithLoading';
+import PipelineRunTimeArgsCounter from 'components/PipelineDetails/PipelineRuntimeArgsCounter';
+import {connect} from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+
+class PipelineRunTimeArgsAllRuns extends PureComponent {
+
+  static propTypes = {
+    runtimeArgs: PropTypes.object,
+    onClose: PropTypes.func.isRequired
+  };
+
+  state = {
+    saving: false,
+    savingAndRunBtnDisabled:this.isRuntimeArgsFilled(this.props.runtimeArgs),
+    savingAndRun: false,
+    error: null
+  };
+
+  componentWillReceiveProps(nextProps) {
+    let {runtimeArgs} = nextProps;
+    this.setState({
+      savingAndRunBtnDisabled: this.isRuntimeArgsFilled(runtimeArgs)
+    });
+  }
+
+  isRuntimeArgsFilled(runtimeArgs) {
+    return runtimeArgs
+      .pairs
+      .filter(runtimearg => !isEmpty(runtimearg.key) && isEmpty(runtimearg.value))
```

Comment:
Maybe change this to `isEmpty(runtimearg.key) || isEmpty(runtimearg.value)`? Right now if I have a value but don't have a key, the 'Save and Run' btn is still enabled